test/test_optim.py

```diff
@@ -1606,25 +1615,31 @@ def closure():
             self.assertEqual(type(res1), type(res2))
 
     @onlyCPU
-    @skipIfTorchDynamo("https://github.com/pytorch/pytorch/issues/123238")
-    def test_grad_scaling_autocast_fused_optimizers(self):
+    @optims([optim for optim in optim_db if "fused" in optim.supported_impls], dtypes=[torch.float32])
+    def test_grad_scaling_autocast_fused_optimizers(self, device, dtype, optim_info):
         # This ut is from test_cuda.py test_grad_scaling_autocast_fused_optimizers
         # but only test Adam/AdamW on CPU
         # TODO: haozhe, support SGD and unified this ut with the CUDA only one
-        optimizer_ctor = (torch.optim.Adam, torch.optim.AdamW)
-        optimizer_kwargs = ({"fused": True, "amsgrad": False}, {"fused": True, "amsgrad": True})
-        separate_unscale = (False, True)
-        for optim, kwargs, _separate_unscale in list(product(optimizer_ctor, optimizer_kwargs, separate_unscale)):
-            self._grad_scaling_autocast_fused_optimizers(
-                optimizer_ctor=optim, optimizer_kwargs=kwargs, separate_unscale=_separate_unscale)
+        if device not in optim_info.supports_fused_on:
+            self.skipTest(f"{device} is not supported for fused on {optim_info.optim_cls.__name__}")
+        optim_inputs = optim_info.optim_inputs_func(device=device)
+        optim_cls = optim_info.optim_cls
+        for optim_input in optim_inputs:
+            kwargs = optim_input.kwargs
+            for _separate_unscale in (True, False):
+                self._grad_scaling_autocast_fused_optimizers(
+                    optimizer_ctor=optim_cls, optimizer_kwargs=kwargs, separate_unscale=_separate_unscale)
 
     def _grad_scaling_autocast_fused_optimizers(self, optimizer_ctor, optimizer_kwargs, separate_unscale):
         (
             mod_control, mod_scaling, opt_control, opt_scaling, data, loss_fn, _,
         ) = _create_scaling_case(optimizer_ctor=optimizer_ctor, optimizer_kwargs=optimizer_kwargs, device='cpu')
         kwargs = deepcopy(optimizer_kwargs)
         kwargs["fused"] = False
-        opt_control = optimizer_ctor(mod_control.parameters(), lr=1.0, **kwargs)
+        if 'lr' not in optimizer_kwargs:
+            # _create_scaling_case will set lr = 1.0 if optimizer_kwargs do not set lr
+            kwargs['lr'] = 1.0
```

Comment:
This is a really large lr--why?

Reply:
It is set here https://github.com/pytorch/pytorch/blob/ec608a5d66398604ec6faf5251a6808f43aace1c/torch/testing/_internal/common_cuda.py#L248-L250 in `_create_scaling_case`. I do not know why it is `1.0`.
I set it to `1.0` here is aim to keep align with `_create_scaling_case`